rust/otap-dataflow/crates/otlp/src/otlp_batch_processor.rs

```diff
@@ -1825,14 +1825,19 @@ mod integration_tests {
         ProcessorWrapper::local(processor, node_config, &config)
     }
 
-    // Helper: Write string to a file
-    fn log_to_file(s: &str) {
-        let mut f = OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open("/tmp/generic_batch_proc_test.json")
-            .expect("could not open /tmp file for writing");
-        writeln!(f, "{s}\n").expect("Write failed");
+    // Helper: Write string to a file (async version)
+    async fn log_to_file(s: &str) {
+        let s = s.to_string();
+        tokio::task::spawn_blocking(move || {
+            let mut f = OpenOptions::new()
+                .create(true)
+                .append(true)
+                .open("/tmp/generic_batch_proc_test.json")
+                .expect("could not open /tmp file for writing");
+            writeln!(f, "{s}\n").expect("Write failed");
+        })
+        .await
+        .expect("Blocking task failed");
```

Comment:
Nice, looks like this will work!

Also for future reference tokio also has an `fs` module that provides a mostly equivalent API to `std::fs`. You might be able to avoid spawning this blocking task by doing:
```suggestion
    async fn log_to_file(s: &str) {
        let mut f = tokio::fs::OpenOptions::new()
            .create(true)
            .append(true)
            .open("/tmp/generic_batch_proc_test.json")
            .await
            .expect("could not open /tmp file for writing");

        _ = f.write(&format!("{s}\n").encode_to_vec()).await.expect("write failed");
    }
```

Reply:
I don't have a strong preference, I think we can merge this as is :) 

Generally, I think people prefer to use the `tokio::fs` approach because you get the the slightly nicer semantics of working with futures vs just having the join handle to the `spawn_blocking` operation. That said, there's some overhead, with the `tokio::fs` module there is some extra overhead of shoehorning the underlying synchronous IO operations into something that is async, so for really performance critical code using `spawn_blocking` can be preferred. This video has a pretty good discussion about the trade-offs https://youtu.be/o2ob8zkeq2s?si=8M7aBM2zXBJ4YM9g&t=5301